app/validators/blocks/list_collector_validator.py

```diff
@@ -104,16 +109,25 @@ def validate_other_list_collectors(self):
                     list_name=list_name,
                 )
 
-    def validate_single_repeating_blocks_list_collector(self):
-        if self.block.get("repeating_blocks"):
-            list_name = self.block["for_list"]
-
-            other_list_collectors = self.questionnaire_schema.get_other_blocks(
-                self.block["id"], for_list=list_name, type="ListCollector"
+    def validate_repeating_blocks_list_collectors(self, same_type, other_type):
+        if not self.block.get("repeating_blocks"):
+            return
+        list_name = self.block["for_list"]
+        section_id = self.questionnaire_schema.get_section_id_for_block(self.block)
+        list_collectors_other_type = self.questionnaire_schema.get_other_blocks(
+            self.block["id"], for_list=list_name, type=other_type
+        )
+        other_list_collectors_same_type = self.questionnaire_schema.get_other_blocks(
+            self.block["id"], for_list=list_name, type=same_type
+        )
+        if [
+            list_collector
+            for list_collector in list_collectors_other_type
+            + other_list_collectors_same_type
+            if list_collector["id"]
+            in self.questionnaire_schema.get_section_block_ids(section_id)
+        ]:
```

Comment:
I find this whole thing pretty hard to read. This definitely need comments.

Why do we get same type and other type?. It is always passing two?.

Reply:
It's removed now since we want to check for multiple list collector on section level, so I reverted it back to its original state when introduced by Kyle.